File: demo/src/main/java/_pfdemo/com/demo/controller/MessageController.java
package _pfdemo.com.demo.controller;

import _pfdemo.com.demo.model.Message;
import _pfdemo.com.demo.service.MessageService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/messages")
public class MessageController {
    private final MessageService messageService;

    public MessageController(MessageService messageService) {
        this.messageService = messageService;
    }

    @GetMapping
    public List<Message> getAllMessages() {
        return messageService.getAllMessages();
    }

    @PostMapping
    public ResponseEntity<Message> addMessage(@RequestBody String content) {
        Message message = messageService.addMessage(content);
        return ResponseEntity.ok(message);
    }
}

File: kotlindemo/src/main/kotlin/kt/example/kotlindemo/controller/MessageController.kt
package kt.example.kotlindemo.controller

import kt.example.kotlindemo.model.Message
import kt.example.kotlindemo.service.MessageService
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/messages")
class MessageController(private val messageService: MessageService) {

    private val logger = LoggerFactory.getLogger(MessageController::class.java)

    @PostMapping
    fun addMessage(@RequestBody message: Message): Message {
        logger.info("Adding new message: ${message.content}")
        return messageService.saveMessage(message)
    }

    @GetMapping
    fun getMessages(): List<Message> {
        logger.info("Fetching all messages")
        return messageService.getAllMessages()
    }
}
